include/boost/numeric/ublas/tensor/static_extents.hpp

```diff
@@ -164,4 +165,18 @@ struct static_product< basic_static_extents<ExtentsType> >{
 
 } // namespace boost::numeric::ublas
 
+namespace boost::numeric::ublas::detail{
+    
+template <class T, T... E>
+struct is_extents< basic_static_extents<T, E...> > : std::true_type {};
+
+template <class T, T... E>
+struct is_static< basic_static_extents<T, E...> > : std::true_type {};
+
+template <class T, std::size_t... E>
+struct is_static_rank< basic_static_extents<T, E...> > : std::true_type {};
+
+} // namespace boost::numeric::ublas
```

Comment:
1. I recommend not to put in `detail` namespace.
2. Could we place it in a file called `extents_traits.hpp` with all the other extents types ?

Reply:
1. Ok
2. same as above